### DISCUSSION TITLE
One part of warm up not working?

### DISCUSSION THREAD

**Discussion by gpb3037** [earthlab-education/Earth-Analytics-AY24#351] [Q&A]
For the section "Add values to a list of converted monthly temperatures"

I was able to run everything in the warm up example but did not get the last flag to clear

my list is 
[0.0, 2.000000000000001, 5.0, 9.6, 5.0, 19.6, 22.8, 21.7, 16.7, 11.6, 4.9, 1.0]
(and if I total them I get
129.3)

reports back
✅ Nice work! The object you called at the end of the cell is of type - list
✅ Nice work! Your list has the correct number of elements
❌ The list you created does not contain the correct values.

that code says
if round(sum(student_temp_long), 1) == 129.3:
    print(("\u2705 The list you created has the correct values in it, good "
           "job!"))
else:
    print("\u274C The list you created does not contain the correct values.")

not sure why I get the else triggered?

**Comment by eculler** [comment]
Hey @gpb3037 Nate mentioned you were having this problem. Normally I'd say this is a problem with the test, but it's working for me and @nquarder, so maybe not. I wouldn't recommend spending a lot of time getting the tests to work -- you're clearly doing the correct calculation here. I'm not totally sure why this is happening, but here's my best guess without watching you run the code:

The test cells in this notebook are designed to be run *immediately* after the cell they are testing. The `_` you see in most of the test cells is a special variable that Jupyter notebooks (.ipynb files) keep that stores the output of the cell that was run immediately previous. If I had to guess...I'd say maybe something didn't run, and the list from the previous cell (unconverted temp values) got passed to the test. A quick thing you can try is to `Restart` the kernel and then `Run All` the cells from the top. You'll find buttons to do that in the top bar above your notebook in VSCode, just to the right of `+ Code` and `+ Markdown`.

If that doesn't work, let's chalk it up to a mystery of the universe and move on!

**Reply by gpb3037** [reply]
might be a mystery -  as I have restarted several times and tried running both each section and the run all.

**Comment by nquarder** [comment]
Hi @gpb3037 - I'm looking at your WARM UP notebook (https://github.com/earthlab-education/climate-coding-challenge-gpb3037/blob/main/WARM%20UP%3A%20Python%20as%20a%20Calculator.ipynb) and it looks like at the end of the cell where you append April - Dec to the Jan-March data you also ask it to print the total (`print(ttotal)`). I'm wondering if having this total printed at the end of the cell is throwing off the test cell? Maybe try removing that print statement and instead print the list of temperatures (`print(temp)`) and rerunning the cell.

I'm including a screenshot of your notebook so you can see where I am talking about: 
<img width="923" alt="Screenshot 2024-09-17 at 9 19 21 AM" src="https://github.com/user-attachments/assets/05a32302-a075-4a24-b92a-956146702576">

Like @eculler mentioned, the test cells are often designed to look at the output of the cell immediately above it. In this case, the test cell is looking for a list of converted temperature values. It is seeing that list of values but it is also seeing the printed total and is getting confused - this is my guess anywho. Again, @gpb3037 try removing that `print(ttotal)` statement and then re-run. Let us know how this goes for you! :) 

One other thing I'll point out is I really like using `.extend()` rather than `.append()` when trying to add items to a list. It seems to be more efficient and can be done in fewer lines of code. Give it a try sometime if you haven't already.  -NQ

**Reply by gpb3037** [reply]
finally found the bug - appended mar twice instead of may.  probably better to add things to an array somehow rather than step by step variables but that was part of the exercise.

**Reply by eculler** [reply]
Yep, this exercise is meant to give you lots of practice with variables and lists, rather than teach you the best way to do things in Python. In general, the warm up notebooks will be more rote practice, which some people find helpful and others find veeery boring. That's why they're optional! The same goes for the extra challenges in the notebooks -- some people will find them motivating and other will find them frustrating. It's totally ok to pick what works for you.

Nice debugging! It's really hard to see one-character differences -- perhaps a good reason to make sure you don't have variables that are very similar to each other. I'm a little confused because it looks like you are actually sending a string to the test cell instead of your list...but if it's passing the tests I'm probably missing something. We aren't planning to do a bunch of test code this semester, so I'm not too worried about that working, though.
